ENI: Trigger NodeManager.Resync immediately after new ENI successfully created

wu0407 opened the pull request:
Current successfully call createInterface not trigger n.manager.resyncTrigger.
It cause spec.ipam.pool update delay, so agent not able allocate ip for new pod
until periodic trigger NodeManager.Resync.

This patch trigger NodeManager.Resync immediately to update ciliumNode resource.

Fixes: #20595
Fixes: bcdb763ca31cd (Fix for excess IP release race condition | handshake between agent and operator)

Signed-off-by: xiaoqing <xiaoqingnb@gmail.com>

Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [x] Thanks for contributing!

<!-- Description of change -->

Fixes: #20595

```release-note
Optimize Eni update latency after new eni created
```

gandro commented on pkg/ipam/node.go:
```diff
@@ -469,7 +469,7 @@ func (n *Node) ResourceCopy() *v2.CiliumNode {
 // of secondary IPs are assigned to the interface up to the maximum number of
 // addresses as allowed by the instance.
-func (n *Node) createInterface(ctx context.Context, a *AllocationAction) error {
+func (n *Node) createInterface(ctx context.Context, a *AllocationAction) (bool, error) {
```
nit: Could you give this boolean return value a name, e.g. `instanceMutated`? It's not clear without context.

Thanks!

wu0407 committed:
Trigger NodeManager.Resync immediately after new ENI successfully created

Current successfully call createInterface not trigger n.manager.resyncTrigger.
It cause spec.ipam.pool update delay, so agent not able allocate ip for new pod
until periodic trigger NodeManager.Resync.

This patch trigger NodeManager.Resync immediately to update ciliumNode resource.

Fixes: #20595
Fixes: bcdb763ca31cd (Fix for excess IP release race condition | handshake between agent and operator)

Signed-off-by: xiaoqing <xiaoqingnb@gmail.com>

gandro reviewed (approved):
Thank you!
